pages/api/review-admin.ts

```diff
@@ -0,0 +1,49 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import mongoose from "mongoose"
+import { ReviewModel } from "../../models/review";
+import { UserModel } from "../../models/user";
+import { getServerSession } from "next-auth";
+import { authOptions } from "./auth/[...nextauth]";
+
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse,) {
+    const session = await getServerSession(req, res, authOptions)
+    await mongoose.connect(process.env.MONGODB_URL as string)      //do I need to connect to database again after this? //also do i need to close connection?
```

Comment:
I think you should probably have one global server MongoDB connection so you're not opening one on each request (and yes doing it like this you should be closing them).

Reply:
Like just this one at the start of the function, and then close it after the function, or would that still open one on each request? Also I was getting a warning from MongoDB saying I'm nearing the connection limit (exceeded 500 for the cluster), is that because I don't close the connections or because of the polling? (Also why do I need to do this in a seperate PR?)

Reply:
Not one at the start of the function (that would still open one on each request), one globally for the entire application that connects without needing a request and then all requests use that connection. I'm not surprised MongoDB is warning you about the connection limit, this is because you don't close the connections and the polling (opening connections frequently in the background) makes the problem worse. I was suggesting doing it in a different PR because it's kind of unrelated to this problem - it's good practice to keep completely separate things separate.